core/src/main/java/org/elasticsearch/index/query/QueryParseContext.java

```diff
@@ -37,10 +37,9 @@
 
     private final IndicesQueriesRegistry indicesQueriesRegistry;
 
-    public QueryParseContext(IndicesQueriesRegistry registry, XContentParser parser, ParseFieldMatcher parseFieldMatcher) {
+    public QueryParseContext(IndicesQueriesRegistry registry, XContentParser parser) {
         this.indicesQueriesRegistry = Objects.requireNonNull(registry, "indices queries registry cannot be null");
         this.parser = Objects.requireNonNull(parser, "parser cannot be null");
-        this.parser.setParseFieldMatcher(Objects.requireNonNull(parseFieldMatcher, "parse field matcher cannot be null"));
```

Comment:
By moving this out we need to make sure the setter is called in many places where we previously implicitly set it via `context.parseFieldMatcher(pfm)`. I only checked the places where previously this method was calles somehow. 
I'm on the fence if maybe we should instead keep it here, that way at least we know there is one location where the parseFieldMatcher is set.